scripts/python/docs-automation/sync_tocs.py

```diff
@@ -8,13 +8,87 @@
 import fnmatch
 import re
 import json
+import pdb
 
 # by default, yaml does not maintain insertion order of the dicts
 # given that this is intended to generate TABLE OF CONTENTS values,
 # maintaining this order is important.
 # The drop-in replacement oyaml is a handy solution for us.
 import oyaml as yaml
 
+MONIKER_REPLACEMENTS = ['{moniker}','<moniker>']
+
+class PathResolver:
+    def __init__(self, doc_repo_location = None, readme_suffix = "", moniker = ""):
+        self.readme_suffix = readme_suffix
+        self.excluded_href_paths = []
+        self.target_moniker = moniker
+
+        self.doc_repo_location = doc_repo_location
+
+        if self.doc_repo_location:
+            self.excluded_href_paths = self.get_non_standard_hrefs(self.doc_repo_location)
+
+    # the doc builds have the capability to reference readmes from external repos (they resolve during publishing)
+    # this means that we can't simply check the href values for existence. If they are an href that STARTS with one of the
+    # "dependent repositories" than we should leave them exactly as is.
+    # amend_href is the core of the logic for handling referenced files and ensures that we cannot refer to the same readme twice
+    # from two different reference ymls
+    def amend_href(self, toc_dict):
+        if not self.doc_repo_location:
+            return toc_dict
+
+        # We want the readme suffix only if we're dealing with a case that does NOT have a target_moniker
+        # this will maintain backwards compatibility with the first method of invoking this script. We don't
+        # want to entirely remove the "suffix" concept though, as the trailing `.pre` is used to update the uid 
+        # on the preview ToC.ymls.
+        # After a moniker folder is introduced, we NO LONGER want the suffix on the readme to change.
+        suffix = "-" + self.readme_suffix + ".md" if self.readme_suffix and not self.target_moniker else  ".md"
+        input_string = toc_dict["href"]
+
+        # if this is an external readme, we should not attempt to resolve the file to a different one, just return with no changes
+        if any([input_string.startswith(href) for href in self.excluded_href_paths]):
+            return toc_dict 
+
+        # create a resolvable path to the readme on disk, without any of the docs ms specificity
+        resolvable_path = os.path.normpath(os.path.join(self.doc_repo_location, input_string.replace("~/", "")))
+
+        # apply moniker folder adjustments if necessary
+        if self.target_moniker is not None:
+            for replacement in MONIKER_REPLACEMENTS:
+                # input string maintains leading ~/ necessary for docs. update the moniker folder if it exists
+                input_string.replace(replacement, self.target_moniker)
```

Comment:
Any example here for the match here?

Reply:
[Example here](https://github.com/MicrosoftDocs/azure-docs-sdk-node/commit/8ecf1146bd610c9f29194ab3b6089a6c7d3e786e)